lib/broccoli/ember-app.js

```diff
@@ -904,7 +904,11 @@ class EmberApp {
 
     // TODO: This isn't quite correct (but it does function properly in most cases),
     // and should be re-evaluated before enabling the `MODULE_UNIFICATION` feature
-    this._srcAfterStylePreprocessing = preprocessCss(srcAfterPreprocessTreeHook, '/src/ui/styles', '/assets', options);
+    let srcStylesPath = this._resolveLocal('src/ui/styles');
+
+    if (existsSync(srcStylesPath)) {
+      this._srcAfterStylePreprocessing = preprocessCss(srcAfterPreprocessTreeHook, '/src/ui/styles', '/assets', options);
```

Comment:
With an app that is partially still using some folders in the app directory, a build with preprocessing in the `app/styles` directory (possibly limited by addon trees support at the time) fails without this check.

I'd definitely be interested in a better way to check this rather exists. I brought this in looking at the logic in the existing `_initOptions` method.